packages/plugins/coverage/src/lib/index.ts

```diff
@@ -40,7 +40,7 @@ export default class Coverage {
     });
 
     this.embark.events.on("tests:ready", this.pushDeployedContracts.bind(this));
-    this.embark.events.on("tests:finished", this.produceCoverageReport.bind(this));
+    this.embark.registerActionForEvent("tests:finished", this.produceCoverageReport.bind(this));
```

Comment:
Even though there hasn't been an issue so far, we might want to do the same for `tests:ready`. I could imagine that it's possible `pushDeployedContracts()` will run into similar issues as it's async.

Thoughts?